src/language-html/printer-html.js

```diff
@@ -1060,14 +1064,10 @@ function printEmbeddedAttributeValue(node, originalTextToDoc, options) {
     const value = getValue();
     if (!value.includes("{{")) {
       return printExpand(
-        textToDoc(
-          value,
-          {
-            parser: "css",
-            __isHTMLStyleAttribute: true,
-          },
-          { stripTrailingHardline: true }
```

Comment:
Note this

Reply:
Yes: `lint:typecheck` reported that the last argument `{ stripTrailingHardline: true }` was not needed in multiple cases, and this seems to be right for `textToDoc` within the `printEmbeddedAttributeValue` function block. Prettier then reformats the other lines.

Reply:
I think tests will fail, after we merge #9094

Reply:
tsc thinks that the last argument `{ stripTrailingHardline: true }` is an extra argument in these function calls; When I looked at the code I that this was right.

I tried the change from #9094 and have not seen any test failures so far in tests/html, tests/css, tests/vue, or tests/angular. I will now try running the full test suite with the change from #9094.

Reply:
That's wired

Reply:
I reverted these changes and added `@ts-ignore` comments. I am running out of ideas on my side about these function calls.

Reply:
`textToDoc` takes only 2 arguments there:
https://github.com/prettier/prettier/blob/23004319a01babadc1e7661091538b4b2110047d/src/language-html/printer-html.js#L1048-L1053

Reply:
Ah, it's another function...

Reply:
So it looks like `textToDoc` has 2 meanings in `src/language-html/printer-html.js`:

- The internal `embed` function receives `textToDoc` as an argument and calls `textToDoc` with 4 arguments;
- Internal `printEmbeddedAttributeValue` function has an inner `textToDoc` function that expects 2 arguments

Any explanations why this would be the case and any ideas how we can avoid this kind of confusion?

Reply:
To avoid confusion, use Ctrl+click, or F12, or hover, or Ctrl+hover to see the type and/or definition of what is called.

Reply:
There is no problem with the original implementation, it's  on me, I was not careful enough.

Reply:
I have now renamed `textToDoc` to `attributeTextToDoc` in the `printEmbeddedAttributeValue` function, to avoid getting us tripped up in the future, and removed the extra parameter again (reformatted with Prettier).